src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,123 @@
+
+        nullFieldUsers = new ArrayList<>();
+        nullFieldUsers.add(new User(null, "Bony", "bonybony", 27));
+        nullFieldUsers.add(new User(62345L, null, "davidbekham", 23));
+        nullFieldUsers.add(new User(73L,"Anna", null, 30));
+        nullFieldUsers.add(new User(623L, "Greg", "greg342", null));
+    }
+
+    @Test
+    void register_negativeAge_NotOk() {
+        User bob = new User(51323L, "Bob", "bobobob", -19);
+        assertThrows(RuntimeException.class, () -> registrationService.register(bob));
+    }
+
+    @Test
+    void register_nullFields_NotOk() {
+        for (User user : nullFieldUsers) {
+            assertThrows(RuntimeException.class, () -> registrationService.register(user));
+        }
+    }
+
+    @Test
+    void register_unValidUsers_NotOk() {
+        assertEquals(unValidUsers.get(0), registrationService.register(unValidUsers.get(0)));
+        assertThrows(RuntimeException.class, () -> {
+            registrationService.register(unValidUsers.get(1));
+        });
+        assertThrows(RuntimeException.class, () -> {
+            registrationService.register(unValidUsers.get(2));
+        });
+        assertThrows(RuntimeException.class, () -> {
+            registrationService.register(unValidUsers.get(3));
+        });
+        assertThrows(RuntimeException.class, () -> {
+            registrationService.register(unValidUsers.get(4));
+        });
+        assertThrows(RuntimeException.class, () -> {
+            registrationService.register(unValidUsers.get(5));
+        });
+    }
+
+    @Test
+    void register_fiveValidUsers_Ok() {
+        for (User valid : validUsers) {
+            assertEquals(valid, registrationService.register(valid));
+        }
+    }
+
+    @Test
+    void register_userIsAlreadyExist_NotOk() {
+        User bob = new User(5436L, "Bob", "bobikbobik", 20);
+        registrationService.register(bob);
+        assertThrows(RuntimeException.class, () -> registrationService.register(bob));
+    }
+
+    @Test
+    void register_inputUserIsNull_NotOk() {
+        assertThrows(RuntimeException.class, () -> registrationService.register(null));
+    }
+
+    @Test
+    void register_userAgeIsIllegal_NotOk() {
+        User alice = new User(1124L, "Alice", "alice321", 16);
+        User andriy = new User(5452L, "Andriy", "andriy2000", 185);
+        assertThrows(RuntimeException.class, () -> registrationService.register(andriy));
+        assertThrows(RuntimeException.class, () -> registrationService.register(alice));
+    }
+
+    @Test
+    void register_userPasswordIsUnValid_NotOk() {
+        User john = new User(8678L, "John", "hello", 35);
+        User jeremy = new User(73745L, "Jeremy", "", 23);
+        assertThrows(RuntimeException.class, () -> registrationService.register(john));
+        assertThrows(RuntimeException.class, () -> registrationService.register(jeremy));
+    }
+
+    @Test
+    void register_fewNullFields() {
```

Comment:
you have already test that check it.
if even one field is null it will throw an exception